packages/loopring_v3/contracts/core/iface/ILoopringV3.sol

```diff
@@ -14,11 +14,14 @@ abstract contract ILoopringV3 is Claimable
     event ExchangeStakeDeposited(address exchangeAddr, uint amount);
     event ExchangeStakeWithdrawn(address exchangeAddr, uint amount);
     event ExchangeStakeBurned(address exchangeAddr, uint amount);
-    event SettingsUpdated(uint time);
+    event SettingsUpdated(uint time, uint effectTime);
+    event SettingsApplied(uint time);
 
     // == Public Variables ==
+    uint256 internal constant SETTING_UPDATE_DELAY = 7 days;
     mapping (address => uint) internal exchangeStake;
 
+    uint public nextEffectiveTime;
```

Comment:
this will probably change the storage layout in an incompartible way.

Reply:
Since you have `contract LoopringV3 is ILoopringV3, ReentrancyGuard`, changing any storage layout in IloopringV3 will break upgradabilty.

Reply:
loopringv3 is not deployed using proxy mode

Reply:
because it has no `initialize` function